download-test_readline.rb

```diff
@@ -51,6 +51,8 @@ def run
             loop do
               break if files.empty?
               path, dir, file = files.shift
+              # x,y,z = [].shift => x is nil
+              break if path.nil?
```

Comment:
> It also has a small change to download-test_readline.rb for thread safety.

Good. I think that you find and fix it carefully was difficult. Would you split it to another commit to making better?

Reply:
Done